src/websocket.js

```diff
@@ -94,39 +99,67 @@ class WebSocket extends EventTarget {
   }
 
   get onopen() {
-    return this.listeners.open;
+    return this.onopenCb;
   }
 
   get onmessage() {
-    return this.listeners.message;
+    return this.onmessageCb;
   }
 
   get onclose() {
-    return this.listeners.close;
+    return this.oncloseCb;
   }
 
   get onerror() {
-    return this.listeners.error;
+    return this.onerrorCb;
   }
 
   set onopen(listener) {
-    delete this.listeners.open;
-    this.addEventListener('open', listener);
+    if (listener === null) {
+      if (this.onopenCb !== null) {
+        this.removeEventListener('open', this.onopenCb);
```

Comment:
you need *always* remove the previous event listener, not only when the argument is null

Reply:
It removes the previous listener below as well using `this.onopen = null`